### PR TITLE
[Breaking Changes] Migrated to Typscript

### PR DESCRIPTION
**Pull request by romitkarmakar**

### CODE REVIEW

**Comment by romitkarmakar** [comment]
HTTP Basic Authentication has been fixed in this PR. For reference check: https://en.wikipedia.org/wiki/Basic_access_authentication

File: package.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "casbin-express-authz",
-  "version": "2.0.0",
+  "version": "3.0.0",
   "description": "casbin-express-authz is an authorization middleware for Express",
-  "main": "authz.js",
```
Comment (nodece) [inline package.json:5]: Please add `typings`

File: package.json [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+  "main": "build/authz.js",
   "scripts": {
-    "prepublish": "npm run lint && npm run test",
-    "lint": "eslint \"authz.js\"",
+    "prepublish": "npm run build && npm run lint && npm run test",
```
Comment (nodece) [inline package.json]: I think we should run `lint`, `test`, `build` in turn.

Comment (romitkarmakar) [reply package.json]: I think I have to run ``build`` before ``test`` as test will not run if the ``.ts`` files are not transpiled to ``.js`` files.

Comment (nodece) [reply package.json]: Use `ts-jest` is working.

File: src/authz.ts [hunk 3]
```diff
@@ -33,18 +33,12 @@ class BasicAuthorizer {
 
         try {
             const header: string = this.req.get("Authorization");
-            if (header != undefined) {
```
Comment (nodece) [inline src/authz.ts]: Miss `null`？

Comment (nodece) [reply src/authz.ts]: `if (header) ` is better.

**Comment by nodece** [comment]
Please fix PR error.

**Comment by romitkarmakar** [comment]
PR error fixed

File: package.json (hunk 2 above)
Comment (romitkarmakar) [reply package.json]: Added

File: package.json (hunk 1 above)
Comment (romitkarmakar) [reply package.json:5]: Added typings support

**Comment by nodece** [comment]
lgtm.

**Comment by nodece** [comment]
@romitkarmakar can we support override `Authorizer`？

**Comment by romitkarmakar** [comment]
@nodece this is a good idea. I will work on this issue and add a subsequent PR for custom ``Authorizer``

**Comment by romitkarmakar** [comment]
@nodece can you publish the changes to npm.

**Comment by hsluoyz** [comment]
@nodece @romitkarmakar please also release on GitHub: https://github.com/node-casbin/express-authz/releases

**Pull request merged**